editor/plugins/org.fusesource.ide.wsdl2rest.ui/src/org/fusesource/ide/wsdl2rest/ui/wizard/Wsdl2RestWizard.java

```diff
@@ -0,0 +1,359 @@
+import org.eclipse.core.runtime.IPath;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.NullProgressMonitor;
+import org.eclipse.core.runtime.Status;
+import org.eclipse.jface.dialogs.ErrorDialog;
+import org.eclipse.jface.dialogs.IMessageProvider;
+import org.eclipse.jface.viewers.ISelection;
+import org.eclipse.jface.viewers.IStructuredSelection;
+import org.eclipse.jface.wizard.Wizard;
+import org.eclipse.ui.INewWizard;
+import org.eclipse.ui.ISelectionService;
+import org.eclipse.ui.IWorkbench;
+import org.eclipse.ui.PlatformUI;
+import org.fusesource.ide.camel.editor.utils.MavenUtils;
+import org.fusesource.ide.camel.model.service.core.io.CamelIOHandler;
+import org.fusesource.ide.camel.model.service.core.util.CamelCatalogUtils;
+import org.fusesource.ide.camel.model.service.core.util.CamelMavenUtils;
+import org.fusesource.ide.foundation.core.util.Strings;
+import org.fusesource.ide.wsdl2rest.ui.internal.UIMessages;
+import org.fusesource.ide.wsdl2rest.ui.internal.Wsdl2RestUIActivator;
+import org.fusesource.ide.wsdl2rest.ui.wizard.pages.Wsdl2RestWizardFirstPage;
+import org.fusesource.ide.wsdl2rest.ui.wizard.pages.Wsdl2RestWizardSecondPage;
+import org.jboss.fuse.wsdl2rest.impl.Wsdl2Rest;
+
+/**
+ * @author brianf
+ *
+ */
+public class Wsdl2RestWizard extends Wizard implements INewWizard {
+
+	/**
+	 * Collection of settings used by the wsdl2rest utility.
+	 */
+	final Wsdl2RestOptions options;
+	
+	public static final String DEFAULT_CONFIG_NAME = "rest-camel-context.xml"; //$NON-NLS-1$
+	public static final String DEFAULT_BLUEPRINT_CONFIG_NAME = "rest-blueprint-context.xml"; //$NON-NLS-1$
+	public static final String DEFAULT_SPRINGBOOT_CONFIG_NAME = "rest-springboot-context.xml"; //$NON-NLS-1$
+
+	private Wsdl2RestWizardSecondPage pageTwo;
+	
+	private boolean inTest = false;
+	
+	public Wsdl2RestWizard() {
+		options = new Wsdl2RestOptions();
+	}
+	
+	public Wsdl2RestWizard(Wsdl2RestOptions options) {
+		this.options = options;
+	}
+
+	@Override
+	public void init(IWorkbench workbench, IStructuredSelection selection) {
+		setWindowTitle(UIMessages.wsdl2RestWizardWindowTitle);
+		setNeedsProgressMonitor(true);
+	}
+
+	@Override
+	public boolean performFinish() {
+		try {
+			generate();
+		} catch (Exception e) {
+			Wsdl2RestUIActivator.pluginLog().logError(e);
+			ErrorDialog.openError(
+					getShell(),
+					UIMessages.wsdl2RestWizardErrorWindowTitle,
+					UIMessages.wsdl2RestWizardErrorMessage,
+					new Status(IStatus.ERROR, Wsdl2RestUIActivator.PLUGIN_ID, e.getMessage(), e));
+			return false;
+		}
+		return true;
+	}
+
+	/**
+	 * Uses the Workbench selection service to find the currently selected project and use it as the source. 
+	 * @return IProject
+	 */
+	protected IProject getSelectedProjectFromSelectionService() {
+		ISelectionService ss = PlatformUI.getWorkbench().getActiveWorkbenchWindow().getSelectionService();
+		String projExpID = "org.eclipse.ui.navigator.ProjectExplorer"; //$NON-NLS-1$
```

Comment:
using the Project explorer means that we are providing the menu only on "project explorer view"? Not in "Package explorer" for instance?
https://github.com/jbosstools/jbosstools-fuse/pull/1138#discussion_r180425769